### ISSUE TITLE
Unions not working

### ISSUE THREAD

**Issue by kvablack** [patrick-kidger/jaxtyping#210]
Can't seem to get unions between JAX arrays and NumPy arrays to work. This includes the built-in `ArrayLike` alias. Any idea what's going on?

jax==0.4.28
jaxtyping==0.2.28

```python
from jaxtyping import Num, Array, ArrayLike
import numpy as np

x = np.zeros([5], np.float32)
print(isinstance(x, Num[np.ndarray, "..."]))  # -> True
print(isinstance(x, Num[ArrayLike, "..."]))  # -> False
print(isinstance(x, Num[np.ndarray | Array, "..."]))  # -> False
print(isinstance(x, Num[Array | np.ndarray, "..."]))  # -> False

# Python scalars seem to work fine
print(isinstance(5, Num[ArrayLike, "..."]))  # -> True
```

**Comment by patrick-kidger** [comment]
See https://github.com/patrick-kidger/jaxtyping/issues/73.

TL;DR: don't do `isinstance(..., Union[Foo, Bar])`. You want `isinstance(..., typing.get_args(Union[Foo, Bar]))`. This is a known Python misfeature/bug.

**Comment by kvablack** [comment]
Ahh thanks, that is very unfortunate....

To make sure I understand fully, in #73 you write:

> And so for some weird reason, as A | B is a different object to Union[A, B], here we find that the former is compatible with isinstance checks in the way you're doing

That seems to imply that the A | B syntax *should* work, at least as long as jaxtyping supports it, which it seems like it does since #77 ?

Also, to doubly clarify, both types of Unions should work when using the `@jaxtyped` decorator since you use `typing.get_args` internally?

**Comment by patrick-kidger** [comment]
Yes to everything!

**Comment by kvablack** [comment]
Thanks!
